ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,188 @@
-  is2DArray(coordinates) {
-    try {
-      const parsedCoords = JSON.parse(coordinates)
-      return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
-    } catch (e) {
-      return false
-    }
+}
+
+function validatePoint(point) {
+  if (
+    point.length !== 2 ||
+     (Number.isNaN(Number.parseFloat(point[0])) &&
+      Number.isNaN(Number.parseFloat(point[1])))
+  ) {
+    return { error: true, message: JSON.stringify(point) + ' is not a valid point.'}
+  } else if (
+    point[0] > 180 ||
+    point[0] < -180 ||
+    point[1] > 90 ||
+    point[1] < -90
+  ) {
+    return { error: true, message: JSON.stringify(point) + ' is not a valid point.'}
   }
-  validatePoint(point) {
+  return { error: false, message: ''}
+}
+
+const BaseLine = props => {
+  const { label, geometryKey, setState, unitKey, widthKey, mode  } = props
+  const [currentValue, setCurrentValue] = useState(JSON.stringify(props[geometryKey]))
+  const [baseLineError, setBaseLineError] = useState({
+    error: false,
+    message: '',
+  })
+  const [bufferError, setBufferError] = useState({ error: false, message: ''})
+
+  useEffect(() => {
+    const { geometryKey } = props
+    setCurrentValue(JSON.stringify(props[geometryKey]))
+  }, [props.polygon, props.line])
+
+  function validateListOfPoints(coordinates) {
+    let message = ''
+    const isLine = mode.includes('line')
+    const numPoints = isLine ? 2 : 4
     if (
-      point.length !== 2 ||
-      (Number.isNaN(Number.parseFloat(point[0])) &&
-        Number.isNaN(Number.parseFloat(point[1])))
+      !mode.includes('multi') &&
+      !coordinates.some(coords => coords.length > 2) &&
+      coordinates.length < numPoints
     ) {
-      return JSON.stringify(point) + ' is not a valid point.'
-    } else if (
-      point[0] > 180 ||
-      point[0] < -180 ||
-      point[1] > 90 ||
-      point[1] < -90
-    ) {
-      return JSON.stringify(point) + ' is not a valid point.'
-    }
-    return ''
-  }
-  validateListOfPoints(coordinates) {
-    let message = ''
-    const isLine = this.props.mode.includes('line')
-    let numPoints = isLine ? 2 : 4
-    if (!this.props.mode.includes('multi')) {
-      if (coordinates.some(coords => coords.length > 2)) {
-        message = ''
-      } else if (coordinates.length < numPoints) {
-        message = `Minimum of ${numPoints} points needed for ${
-          isLine ? 'Line' : 'Polygon'
-        }`
-      }
+      message = `Minimum of ${numPoints} points needed for ${
+        isLine ? 'Line' : 'Polygon'
+      }`
     }
-    for (let i = 0; i < coordinates.length; i++) {
-      if (coordinates[i].length > 2) {
-        coordinates[i].forEach(coordinate => {
-          if (this.validatePoint(coordinate)) {
-            message = this.validatePoint(coordinate)
+    coordinates.forEach(coordinate => {
+      if (coordinate.length > 2) {
+        coordinate.forEach(coord => {
+          if (validatePoint(coord)) {
+            message = validatePoint(coord)
```

Comment:
`validatePoint()` now returns an object like `{ error, message }`, but when you assign it to `message` and then return `{ error, message }`, you're getting a nested object

Reply:
something like this instead? 
```
        coordinate.forEach(coord => {
          const { err, msg } = validatePoint(coord)
          if (err) {
            message = msg
          }
        })
```